File: examples/python/core/logger/use-powertools-production.py
from aws_lambda_powertools import Logger

logger = Logger(service="my-service")

@durable_execution
def handler(event: dict, context: DurableContext) -> str:
    context.set_logger(logger)
    # Now you get JSON logs with all Powertools for AWS Lambda (Python) features
    context.logger.info("Processing started")

File: examples/java/core/logger/use-powertools-production.java
// Java equivalent: use ctx.getLogger() for production logging.
// It provides JSON-structured logs with automatic MDC enrichment
// and replay-aware deduplication — no extra setup needed.

public class ProductionHandler extends DurableHandler<Map<String, Object>, String> {

    @Override
    public String handleRequest(Map<String, Object> event, DurableContext ctx) {
        // Structured JSON logs with all SDK context fields
        ctx.getLogger().info("Processing started");
        // ...
    }
}
